### ISSUE TITLE
Master fails CI

### ISSUE THREAD

**Issue by mouse07410** [openssl/openssl#4103]
Failure appears to be related to Python interface. Here's [the failing build](https://travis-ci.org/mouse07410/openssl/jobs/261629718). I tried that same build twice, same result:
```
. . . . .
==================================== ERRORS ====================================
____ ERROR at setup of TestPKCS7.test_invalid_padding[128-] ____

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestPKCS7::()::test_invalid_padding[128-\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
__ ERROR at teardown of TestPKCS7.test_invalid_padding[128-] ___

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestPKCS7::()::test_invalid_padding[128-\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
__ ERROR at setup of TestANSIX923.test_invalid_padding[128-] ___

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_invalid_padding[128-\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
_ ERROR at teardown of TestANSIX923.test_invalid_padding[128-] _

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_invalid_padding[128-\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
___ ERROR at setup of TestANSIX923.test_pad[128-1111111111-1111111111] ___

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-1111111111-1111111111\x00\x00\x00\x00\x00\x06] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
_ ERROR at teardown of TestANSIX923.test_pad[128-1111111111-1111111111] __

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-1111111111-1111111111\x00\x00\x00\x00\x00\x06] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at setup of TestANSIX923.test_pad[128-111111111111111122222222222222-111111111111111122222222222222] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-111111111111111122222222222222-111111111111111122222222222222\x00\x02] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at teardown of TestANSIX923.test_pad[128-111111111111111122222222222222-111111111111111122222222222222] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-111111111111111122222222222222-111111111111111122222222222222\x00\x02] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at setup of TestANSIX923.test_pad[128-1111111111111111-1111111111111111] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-1111111111111111-1111111111111111\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at teardown of TestANSIX923.test_pad[128-1111111111111111-1111111111111111] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-1111111111111111-1111111111111111\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at setup of TestANSIX923.test_pad[128-11111111111111111-11111111111111111] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-11111111111111111-11111111111111111\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0f] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at teardown of TestANSIX923.test_pad[128-11111111111111111-11111111111111111] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_pad[128-11111111111111111-11111111111111111\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0f] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
__ ERROR at setup of TestANSIX923.test_unpad[128-1111111111-1111111111] __

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_unpad[128-1111111111-1111111111\x00\x00\x00\x00\x00\x06] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at teardown of TestANSIX923.test_unpad[128-1111111111-1111111111] _

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_unpad[128-1111111111-1111111111\x00\x00\x00\x00\x00\x06] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at setup of TestANSIX923.test_unpad[128-111111111111111122222222222222-111111111111111122222222222222] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_unpad[128-111111111111111122222222222222-111111111111111122222222222222\x00\x02] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
 ERROR at teardown of TestANSIX923.test_unpad[128-111111111111111122222222222222-111111111111111122222222222222] 

self = {'TRAVIS_PULL_REQUEST_BRANCH': '', 'rvm_version': '1.29.2 (latest)', 'LC_CTYPE': 'en_US.UTF-8', 'RUBY_VERSION': 'ruby-...st/../include -I/home/travis/build/mouse07410/openssl/test/../include -L/home/travis/build/mouse07410/openssl/test/..'}
key = 'PYTEST_CURRENT_TEST'
item = 'tests/hazmat/primitives/test_padding.py::TestANSIX923::()::test_unpad[128-111111111111111122222222222222-111111111111111122222222222222\x00\x02] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

../venv-pycrypto/lib/python2.7/os.py:471: TypeError
============ 83292 passed, 3212 skipped, 16 error in 270.36 seconds ============
../util/shlib_wrap.sh sh recipes/95-test_external_pyca_data/cryptography.sh => 1
not ok 1 - running Python Cryptography tests

#   Failed test 'running Python Cryptography tests'
#   at ../test/recipes/95-test_external_pyca.t line 27.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 

Test Summary Report
-------------------
../test/recipes/95-test_external_pyca.t     (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=3, Tests=3, 561 wallclock secs ( 4.30 usr  1.77 sys + 367.17 cusr 31.22 csys = 404.46 CPU)
Result: FAIL
make[1]: *** [_tests] Error 1
make[1]: Leaving directory `/home/travis/build/mouse07410/openssl'
make: *** [tests] Error 2
+//// MAKE TEST FAILED
. . . . .
```

**Comment by richsalz** [comment]
See https://github.com/pyca/cryptography/issues/3851  Maybe we can fix it, or maybe we wait for them to tweak things.  @mattcaswell you have any ideas?

**Comment by mattcaswell** [comment]
Fixed by #4139
